### PR TITLE
[Doc] Document Filter API

### PR DESCRIPTION
**Pull request by gregkalapos**
Follow up from #792.

### CODE REVIEW

File: docs/public-api.asciidoc [hunk 1]
```diff
@@ -607,3 +607,44 @@ var asyncResult = await span.CaptureSpan("Select FROM customer", ApiConstants.Ty
...
+Each filter callback will be called in the order they were added, and will receive a payload object as the only argument, containing the data about to be sent to the APM Server.
+
+The filter callback is synchronous and should return the manipulated payload object. If a filter callback doesn’t return any value or returns a falsy value, the remaining filter callback will not be called and the payload will not be sent to the APM Server.
+
+There are 3 overloads of the `Agent.AddFilter` method with the following arguments:
+
+- `Func<ITransaction, ITransaction>`: A filter called for every transaction
+- `Func<ISpan, ISpan>`: A filter called for every span
+- `Func<IError, IError>`: A filter called for every error.
+
```
Comment (gregkalapos) [inline docs/public-api.asciidoc:628]: Maybe we could add something here, like 

> Below are some example usages of the `Agent.AddFilter` method. 

Not sure what's the best way to do that.

Comment (bmorelli25) [reply docs/public-api.asciidoc:628]: Your text sounds good to me! Although I might saw two words: `Below are some usage examples of the Agent.AddFilter method.`

File: docs/public-api.asciidoc [hunk 2]
```diff
@@ -607,3 +607,44 @@ var asyncResult = await span.CaptureSpan("Select FROM customer", ApiConstants.Ty
...
 
 NOTE: Code samples above use `Elastic.Apm.Agent.Tracer.CurrentTransaction`. In production code you should make sure the `CurrentTransaction` is not `null`.
+
+[float]
+[[filter-api]]
+== Filter API (added[1.5])
+
+Use `Agent.AddFilter(filter)` to supply a filter callback.
+
+Each filter callback will be called just before data is being sent to the APM Server. This will allow you to manipulate the data being sent, for instance to remove sensitive information like passwords etc.
```
Comment (bmorelli25) [inline docs/public-api.asciidoc:617]:
```suggestion
Each filter callback will be called just before data is sent to the APM Server. This allows you to manipulate the data being sent, like to remove sensitive information such as passwords.
```

File: docs/public-api.asciidoc [hunk 3]
```diff
@@ -607,3 +607,44 @@ var asyncResult = await span.CaptureSpan("Select FROM customer", ApiConstants.Ty
...
+
+Each filter callback will be called in the order they were added, and will receive a payload object as the only argument, containing the data about to be sent to the APM Server.
```
Comment (bmorelli25) [inline docs/public-api.asciidoc:619]:
```suggestion
Each filter callback is called in the order they are added and will receive a payload object containing the data about to be sent to the APM Server as the only argument.
```

File: docs/public-api.asciidoc [hunk 4]
```diff
@@ -607,3 +607,44 @@ var asyncResult = await span.CaptureSpan("Select FROM customer", ApiConstants.Ty
...
+
+The filter callback is synchronous and should return the manipulated payload object. If a filter callback doesn’t return any value or returns a falsy value, the remaining filter callback will not be called and the payload will not be sent to the APM Server.
+
+There are 3 overloads of the `Agent.AddFilter` method with the following arguments:
+
+- `Func<ITransaction, ITransaction>`: A filter called for every transaction
+- `Func<ISpan, ISpan>`: A filter called for every span
+- `Func<IError, IError>`: A filter called for every error.
```
Comment (bmorelli25) [inline docs/public-api.asciidoc:627]: Inconsistent period usage here. I recommend adding `.` to all.

**Pull request merged**